preview/MsixCore/MsixCoreInstaller/RegistryDevirtualizer.cpp

```diff
@@ -68,6 +68,54 @@ HRESULT RegistryDevirtualizer::Run(_In_ bool remove)
     return S_OK;
 }
 
+HRESULT RegistryDevirtualizer::DeleteKeyIfPresent(HKEY rootHive, std::wstring subKeyPath, std::wstring extensionName)
+{
+    if (!m_hiveFileNameExists)
+    {
+        return S_OK;
+    }
+    std::wstring rootPath = m_loadedHiveKeyName + L"\\Registry";
+    RETURN_IF_FAILED(m_rootKey.Open(HKEY_USERS, rootPath.c_str(), KEY_READ)); // root key until registry
+
+    // map real root hive to virtual root hive
+    PCWSTR virtualKeyPath = L"";
+    for (auto mapping : mappings)
+    {
+        if (mapping.actualRootHive == rootHive && mapping.actualSubkey == subKeyPath)
+        {
+            virtualKeyPath = mapping.virtualKey;
+        }
+    }

```

Comment:
this approach doesn't really work for when HKLM is the rootHive and the subKeyPath is anything besides L""

Reply:
Modified approach to generate virtual mount path based on subpath passed and not the mapping table.